src/main/scala/uk/gov/hmrc/services/validation/utils/Utils.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.services.validation.utils
+
+import java.time.format.DateTimeFormatter
+
+import uk.gov.hmrc.services.validation.models.Cell
+
+import scala.util.Try
+
+object Utils {
+  def compareCellToRule(regex: Option[String], isDate: Boolean, cellValue: String): Option[Boolean] = {
+    if (cellValue.trim.isEmpty) {
+      None
+    } else if (isDate) {
+      Some(Try(DateTimeFormatter.ofPattern("yyyy-MM-dd").parse(cellValue)).isSuccess)
+    } else {
+      Some(cellValue.matches(regex.get))
+    }
+  }
+
+  def compareCellsToGroupRule(flagValue: String, cellToCheck: String, dependentCellValue: String): Boolean = {
+    if (cellToCheck.matches("^" + flagValue + "$")) dependentCellValue.nonEmpty else true
+  }
+
+  def failsMandatoryCheck(isMandatory: Boolean, cell: Cell): Boolean = {
+    if (isMandatory) cell.value.trim.isEmpty else false
```

Comment:
This is giving me true else false vibes, it's a nitpick but I think it could be shortened to `isMandatory && cell.value.trim.isEmpty`

Reply:
good spots, will clean that up